hoomd/md/update.py

```diff
@@ -149,133 +150,140 @@ class mueller_plathe_flow:
+
+        flow_direction (str): Direction of the flow. Can be "X",
+            "Y", or "Z"
+
+        n_slabs (int): Number of slabs used to divide the simulation box along
+            the shear gradient. Using too few slabs will lead to a larger volume
+            being disturbed by the momentum exchange, while using too many slabs
+            may mean that there are not enough particles to exchange the target
+            momentum.
+
         max_slab (int): Id < n_slabs where the max velocity component is search
             for. If set < 0 the value is set to its default n_slabs/2.
+
         min_slab (int): Id < n_slabs where the min velocity component is search
             for. If set < 0 the value is set to its default 0.
 
-    .. attention::
+    Attention:
         * This updater has to be always applied every timestep.
         * This updater works currently only with orthorhombic boxes.
 
-    .. note:
-        If you set this updater with unrealistic values, it algorithm might not
-        terminate, because your desired flow target can not be achieved.
-
-    .. versionadded:: v2.1
 
     Examples::
 
-        #const integrated flow with 0.1 slope for max 1e8 timesteps
-        const_flow = hoomd.variant.linear_interp( [(0,0),(1e8,0.1*1e8)] )
-        #velocity gradient in z direction and shear flow in x direction.
-        update.mueller_plathe_flow(
-            all, const_flow, md.update.mueller_plathe_flow.Z,
-            md.update.mueller_plathe_flow.X, 100)
+        # const integrated flow with 0.1 slope for max 1e8 timesteps
+        ramp = hoomd.variant.Ramp(0.0, 0.1e8, 0, int(1e8))
+        # velocity gradient in z direction and shear flow in x direction.
+        mpf = hoomd.md.update.ReversePerturbationFlow(filter=hoomd.filter.All(), flow_target=ramp, slab_direction="Z", flow_direction="X", n_slabs=20)
+
+        self._flow_target = hoomd.variant._setup_variant_input(flow_target)
+
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this updater.
+
+        flow_target (hoomd.variant.Variant): Integrated target flow in the
+            natural units of the simulation.
+
+        slab_direction (str): Direction perpendicular to the
+            slabs.
+
+        flow_direction (str): Direction of the flow.
+
+        n_slabs (int): Number of slabs.
+
+        max_slab (int): Id < n_slabs where the max velocity component is
+            searched for.
 
+        min_slab (int): Id < n_slabs where the min velocity component is
+            searched for.
     """
 
     def __init__(self,
-                 group,
+                 filter,
                  flow_target,
                  slab_direction,
                  flow_direction,
                  n_slabs,
                  max_slab=-1,
                  min_slab=-1):
-        period = 1  # This updater has to be applied every timestep
+
         assert (n_slabs > 0), "Invalid negative number of slabs."
```

Comment:
Switch these to use `if` checks and raise `ValueError` or other appropriate exceptions. `assert` is meant to be used to check for errors in the implementation, not validate user parameters.

Reply:
This has not been resolved.